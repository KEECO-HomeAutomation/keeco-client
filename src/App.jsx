import React from 'react';
import {
	compose,
	mapProps,
	onlyUpdateForKeys,
	branch,
	renderComponent
} from 'recompose';
import { ApolloProvider } from 'react-apollo';
import { Provider, connect } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import { MuiThemeProvider } from '@material-ui/core/styles';
import getApolloClient from './data/apolloClient';
import getReduxStore from './data/reduxStore';
import theme from './style/theme';

import Scenes from './scenes';

const redux = getReduxStore();

const EnhancedApolloProvider = compose(
	connect(state => ({
		serverPath: state.app.server.path
	})),
	onlyUpdateForKeys(['serverPath']),
	branch(
		({ serverPath }) => serverPath === null,
		renderComponent(({ children }) => children)
	),
	mapProps(({ serverPath, children }) => ({
		client: getApolloClient(
			serverPath,
			() => redux.store.getState().app.login.token,
			() => redux.store.dispatch({ type: 'LOGIN@CLEAR_TOKEN' })
		),
		children
	}))
)(ApolloProvider);

const App = () => {
	return (
		<Provider store={redux.store}>
			<PersistGate persistor={redux.persistor}>
				{bootstrapped => {
					if (bootstrapped) {
						return (
							<EnhancedApolloProvider>
								<MuiThemeProvider theme={theme}>
									<Scenes />
								</MuiThemeProvider>
							</EnhancedApolloProvider>
						);
					} else {
						return <p>Loading...</p>;
					}
				}}
			</PersistGate>
		</Provider>
	);
};

export default App;
