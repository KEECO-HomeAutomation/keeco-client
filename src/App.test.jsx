import React from 'react';
import { mount } from 'enzyme';

jest.mock('./scenes', () => () => <p>Scenes</p>);

import App from './App';

describe('<App />', () => {
	test('Renders without crashing', () => {
		const comp = mount(<App />);
		expect(comp.length).toBe(1);
	});

	test('Should match snapshot', () => {
		expect(mount(<App />)).toMatchSnapshot();
	});
});
