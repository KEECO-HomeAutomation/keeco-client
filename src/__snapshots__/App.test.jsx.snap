// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should match snapshot 1`] = `
<div>
  <h1>
    KEECO Client
  </h1>
  <input
    defaultChecked={true}
    type="checkbox"
  />
</div>
`;
